helix-view/src/document.rs

```diff
@@ -1410,6 +1410,7 @@ impl Document {
     }
 
     fn undo_redo_impl(&mut self, view: &mut View, undo: bool) -> bool {
+        self.append_changes_to_history(view);
```

Comment:
```suggestion
        if undo {
            self.append_changes_to_history(view);
        } else if !self.changes.is_empty() {
            return false;
        }
```

and similar for `earlier_later_impl` below, see https://github.com/mawww/kakoune/blob/e0bbd1e7ca7b96531671682bc63c5ee94d1fbc85/src/buffer.cc#L324-L326

Reply:
Thanks for the reference, I'll be sure to look at Kakoune for future issues like this where there's a lot of parallels. Should be fixed now.